### PR TITLE
Fix awk linter and security concerns.

### LINKED ISSUE #1401: Awk linter executes arbitrary code
## Information

**VIM version**
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Feb 28 2018 14:56:56)
Included patches: 1-1553

Operating System: Fedora Linux 27

### :ALEInfo

```
 Current Filetype: awk
Available Linters: ['gawk']
  Enabled Linters: ['gawk']
 Linter Variables:

let g:ale_awk_gawk_executable = 'gawk'
let g:ale_awk_gawk_options = ''
 Global Variables:

let g:ale_cache_executable_check_failures = 0
let g:ale_change_sign_column_color = 0
let g:ale_command_wrapper = ''
let g:ale_completion_delay = 100
let g:ale_completion_enabled = 0
let g:ale_completion_max_suggestions = 50
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '%code: %%s'
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 0
let g:ale_fixers = {}
let g:ale_history_enabled = 1
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = 0
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 0
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'never'
let g:ale_lint_on_insert_leave = 0
let g:ale_linter_aliases = {}
let g:ale_linters = {}
let g:ale_linters_explicit = 0
let g:ale_list_window_size = 10
let g:ale_list_vertical = 0
let g:ale_loclist_msg_format = '%code: %%s'
let g:ale_max_buffer_history_size = 20
let g:ale_max_signs = -1
let g:ale_maximum_file_size = 0
let g:ale_open_list = 1
let g:ale_pattern_options = {}
let g:ale_pattern_options_enabled = 0
let g:ale_set_balloons = 1
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = 0
let g:ale_sign_error = '>>'
let g:ale_sign_info = '--'
let g:ale_sign_offset = 1000000
let g:ale_sign_style_error = '>>'
let g:ale_sign_style_warning = '--'
let g:ale_sign_warning = '--'
let g:ale_statusline_format = ['%d error(s)', '%d warning(s)', 'OK']
let g:ale_type_map = {}
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
  Command History:

(executable check - success) gawk
(finished - exit code 0) ['/bin/zsh', '-c', 'gawk  -f ''/tmp/vYzUS12/1/bad.awk'' --lint /dev/null']

<<<NO OUTPUT RETURNED>>>
```

## What went wrong

Awk scripts are executed with the provided awk checker and awk functions like `system("shell code")` can call arbitrary shell code.

## Reproducing the bug

Steps for repeating the bug:

  * create a file called whatever.awk with a single line like: `BEGIN { system("touch some-file") }`
  * after triggering a lint check, the file some-file will be in your current working directory: `ls -l some-file`

## Workaround

this is fixed by adding:
`let g:ale_awk_gawk_options = "--source 'BEGIN { exit 0 } END { exit 0 }'"`

This terminates the execution of the script before it can run anything, like other BEGIN or END blocks.

### PR DESCRIPTION
**Pull request by adedomin**
closes #1401 

  * Fixes arbitrary code execution flaw.
  * adds handler with unit test; the cpplint one was not working for me.

### CODE REVIEW

**Comment by adedomin** [comment]
Done, let me know if there are any other issues.

note: I made some new changes/improvements.

File: ale_linters/awk/gawk.vim [hunk 1]
```diff
@@ -4,23 +4,23 @@
 let g:ale_awk_gawk_executable =
 \   get(g:, 'ale_awk_gawk_executable', 'gawk')
 
-let g:ale_awk_gawk_options =
-\   get(g:, 'ale_awk_gawk_options', '')
```
Comment (w0rp) [inline ale_linters/awk/gawk.vim]: We should keep these options, so people can set other options.

File: test/command_callback/test_gawk_command_callback.vader [hunk 2]
```diff
@@ -0,0 +1,27 @@
+Before:
```
Comment (w0rp) [inline test/command_callback/test_gawk_command_callback.vader]: Looks good. Add a test here to check that you can still set other options for `gawk`.

**Comment by adedomin** [comment]
not sure how useful the command line options are. but re-added anyhow.

**Comment by w0rp** [comment]
Cheers! :beers:

**Pull request merged**